frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +191,79 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
     500,
   );
 
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
+    const [TypedArrayClass] = getConstructorForElementClass(elementClass);
+
+    const [cuboidXY, cuboidXZ, cuboidYZ] = await Promise.all([
+      api.data.getDataForViewport(OrthoViews.PLANE_XY, layerName),
+      api.data.getDataForViewport(OrthoViews.PLANE_XZ, layerName),
+      api.data.getDataForViewport(OrthoViews.PLANE_YZ, layerName),
+    ]);
+    const dataForAllViewports = new TypedArrayClass(
+      cuboidXY.length + cuboidXZ.length + cuboidYZ.length,
+    );
+
+    dataForAllViewports.set(cuboidXY);
+    dataForAllViewports.set(cuboidXZ, cuboidXY.length);
+    dataForAllViewports.set(cuboidYZ, cuboidXY.length + cuboidXZ.length);
+
+    const localHist = {};
+    for (let i = 0; i < dataForAllViewports.length; i++) {
+      if (dataForAllViewports[i] !== 0 && dataForAllViewports[i] in localHist) {
+        localHist[dataForAllViewports[i]] += 1;
+      } else {
+        localHist[dataForAllViewports[i]] = 1;
+      }
+    }
+
+    const accumulator = {};
+    let area = 0;
+    let lastKey = 0;
+    for (const key of Object.keys(localHist)) {
```

Comment:
For the accumulation it's important to iterate in the correct order over the data structure. Enumeration order is a complex topic in JS, unfortunately. I think, in your use case, it's working correctly with modern browsers, but I'd still want to keep it safe by doing a simple sort beforehand. Something like this:

```
// assuming that localHist is a Map()
const sortedHistKeys = Array.from(localHist.keys()).sort((a, b) => a - b)
for (const key of sortedHistKeys) {
...
```